pkg/controllers/disruption/controller.go

```diff
@@ -79,6 +79,8 @@ func NewController(clk clock.Clock, kubeClient client.Client, provisioner *provi
 		cloudProvider: cp,
 		lastRun:       map[string]time.Time{},
 		methods: []Method{
+			// Terminate any NodeClaims with StatusConditions specific to the CloudProvider implementation. For
+			NewCloudProvider(kubeClient, cluster, c.cloudProvider, recorder, provisioner),
```

Comment:
What about appending `NewStatusConditionDisruption()` with new methods gen-ed based on a combination of the CloudProvider reason and the Drift reason?